narf/combineutils.py

```diff
@@ -386,6 +400,18 @@ def __init__(self, indata, options):
         # observed number of events per bin
         self.nobs = tf.Variable(self.indata.data_obs, trainable=False, name="nobs")
 
+        if self.chisqFit:
+            if self.externalCovariance:
+                if self.indata.data_cov_inv is None:
+                    raise RuntimeError("No external covariance found in input data.")
+                # provided covariance
+                self.data_cov_inv = self.indata.data_cov_inv
+            else:
+                # covariance from data stat
+                if any(self.nobs<=0):
```

Comment:
Sorry I didn't notice this before.  using the python builtin any will trigger an explicit python interpreter loop over the elements of self.nobs.

Most efficient would be ```if tf.math.reduce_any(self.nobs <= 0).numpy()```